plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, List, NamedTuple, Optional, Union
+
+import duckdb
+import pandas as pd
+import pyarrow as pa
+
+from flytekit import PythonInstanceTask
+from flytekit.extend import Interface
+from flytekit.types.structured.structured_dataset import StructuredDataset
+
+
+class QueryOutput(NamedTuple):
+    counter: int = -1
+    output: Optional[str] = None
+
+
+class DuckDBQuery(PythonInstanceTask):
+    _TASK_TYPE = "duckdb"
+
+    def __init__(
+        self,
+        name: str,
+        query: Union[str, List[str]],
+        inputs: Optional[Dict[str, Union[StructuredDataset, list]]] = None,
+        **kwargs,
```

Comment:
Could we add an Output Schema type here like [snowflake](https://github.com/flyteorg/flytekit/blob/707fc03b88ec08fb10d60ac642f3c80e275bd90c/plugins/flytekit-snowflake/flytekitplugins/snowflake/task.py#L45)? if output_schema_type is none, we won't generate output dataset. 
And the type should change to StructuredDataset, because we already deprecated FlyteSchema

Reply:
@pingsutw, this task isn't helpful if there's no output dataset. The `DuckDBQuery` task runs some queries and returns the output of a `SELECT` statement. Hence, it must return the query output, and in this case, it's `StructuredDataset`. Also, I can definitely add `output_schema_type`. But the output has to always be a `StructuredDataset`. So is it necessary? I'm already hard-coding the output type in the initialization. Let me know what you think.

Reply:
I see. do we support `insert` or some other operations? If not, I think we don't need schema type for now

Reply:
Yeah, we do. So you can give a bunch of queries to a single `DuckDBQuery` task. But the last one needs to be a SELECT query because after say, you insert the data, you need to retrieve the data, right? Else, it's of no use. I'm using the non-persistent offering by DuckDB. So all the data will be available only within the query. Does that make sense?

Reply:
make sense. Thanks for the explanation.